homeassistant/components/schlage/switch.py

```diff
@@ -0,0 +1,123 @@
+"""Platform for Schlage switch integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import partial
+from typing import Any
+
+from pyschlage.lock import Lock
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SchlageDataUpdateCoordinator
+from .entity import SchlageEntity
+
+
+@dataclass
+class SchlageSwitchEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    # NOTE: This has to be a mixin because these are required keys.
+    # SwitchEntityDescription has attributes with default values,
+    # which means we can't inherit from it because you haven't have
+    # non-default arguments follow default arguments in an initializer.
+
+    on_fn: Callable[[Lock], None]
+    off_fn: Callable[[Lock], None]
+    value_fn: Callable[[Lock], bool | None]
```

Comment:
At what point can the value be none?

Reply:
It can't, but I have an incorrect annotation in the pyschlage library: https://github.com/dknowles2/pyschlage/blob/main/pyschlage/lock.py#L46

Even when creating the object it's always a bool: https://github.com/dknowles2/pyschlage/blob/main/pyschlage/lock.py#L106

(Same goes for `lock_and_leave` as well)

I made this just return a bool; I don't think pytype is strict/smart enough to catch the inconsistency.

Reply:
Ah I thought Mypy would complain but he doesn't, that's nice.

Reply:
It would be nice if the `| None` could be removed, but that can be done later as well

Reply:
> Ah I thought Mypy would complain but he doesn't, that's nice.

Yeah I was also surprised. I think it gets confused by lambdas? 🤷🏻‍♂️ 

> It would be nice if the `| None` could be removed, but that can be done later as well

I did that in https://github.com/home-assistant/core/pull/98004/commits/5aba4492ea66c670d0b104e5f91c72556ddef632 😃

Reply:
https://github.com/dknowles2/pyschlage/pull/82